### PR TITLE
Implement Client Connector and Samples

### PR DESCRIPTION
**Pull request by jaymine**

### CODE REVIEW

File: component/samples/ethereum/sampleDashBoard.bal [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import ballerina.net.http;
+import ballerina.util;
```
Comment (anupama-pathirage) [inline component/samples/ethereum/sampleDashBoard.bal]: Why do we need util here?

File: component/src/org/wso2/ballerina/connectors/ethereum/ClientConnector.bal [hunk 2]
```diff
@@ -0,0 +1,706 @@
...
+const string ETH_GET_FILTER_CHANGES = "eth_getFilterChanges";
+const string ETH_GET_FILTER_LOGS = "eth_getFilterLogs";
+const string ETH_GET_LOGS = "eth_getLogs";
+const string ETH_GET_WORK = "eth_getWork";
+const string ETH_SUBMIT_WORK = "eth_submitWork";
+const string ETH_SUBMIT_HASHRATE = "eth_submitHashrate";
+const string EMPTY_PARAMS = "[]";
+
+@Description {value:"Ethereum client connector"}
+@Param {value:"URI: URI of the Ethereum JSON - RPC server"}
```
Comment (anupama-pathirage) [inline component/src/org/wso2/ballerina/connectors/ethereum/ClientConnector.bal]: Add doc comments for all the parameters

File: component/src/org/wso2/ballerina/connectors/ethereum/ClientConnector.bal [hunk 3]
```diff
@@ -0,0 +1,706 @@
...
+public connector ClientConnector (string URI, http:Options options, string jsonRPCVersion, int id) {
+
+    endpoint<http:HttpClient> ethereumEP {
+        create http:HttpClient(URI, options);
+    }
+
+    @Description {value:"Returns the current client version"}
+    @Param {value:"jsonRPCVersion: JSON RPC version"}
+    @Param {value:"id: Network id"}
```
Comment (anupama-pathirage) [inline component/src/org/wso2/ballerina/connectors/ethereum/ClientConnector.bal]: id and jsonRPCVersion are no longer params for this action.

File: component/src/org/wso2/ballerina/connectors/ethereum/ClientConnector.bal [hunk 4]
```diff
@@ -0,0 +1,706 @@
...
+    @Return {value:"InResponse object"}
+    @Return {value:"Error occured during HTTP client invocation"}
+    action web3ClientVersion () (json, http:HttpConnectorError) {
+        var response, e = ethereumEP.post("/", constructRequest(jsonRPCVersion, id, WEB3_CLIENT_VERSION, EMPTY_PARAMS));
+        return response.getJsonPayload(), e;
+    }
+
+    @Description {value:"Returns Keccak-256 (not the standardized SHA3-256) of the given data"}
+    @Param {value:"jsonRPCVersion: JSON RPC version"}
+    @Param {value:"id: Network id"}
```
Comment (anupama-pathirage) [inline component/src/org/wso2/ballerina/connectors/ethereum/ClientConnector.bal]: id and jsonRPCVersion are no longer params for this action. Check all the actions

File: component/samples/ethereum/sample.bal [hunk 5]
```diff
@@ -0,0 +1,224 @@
+//
+// Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (anupama-pathirage) [inline component/samples/ethereum/sample.bal]: Some files are in 2017 while others in 2018. Is that correct?

File: README.md [hunk 6]
```diff
@@ -1 +1,2326 @@
...
+    `bin$ ballerina run sample.bal <URI> <JSONRPCVersion> <NetworkID> <MethodName> <Param1> .. <ParamN>`
+
+## Working with Ethereum connector actions
+
+### Example
+
+All the actions return a `JSON payload` and a `http:HttpConnectorError`. The `JSON payload` consists of either a result or an error. The following section provides details on the result of the `JSON` payload for each action under the **Returns** section.
+
+
+```java
```
Comment (anupama-pathirage) [inline README.md]: Use ```ballerina for ballerina codes

**Review by anupama-pathirage — CHANGES_REQUESTED** [review]

File: component/samples/ethereum/sampleDashBoard.bal [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    println("------------------------------------------------------");
+
+    //get client version
+    var response, e = ethereumConnector.web3ClientVersion();
+    if (e != null) {
+        println(e);
+        return;
+    }
+    println("Web3 Client Version: " + response.result.toString());
+
```
Comment (anupama-pathirage) [inline component/samples/ethereum/sampleDashBoard.bal]: Shall we  have  a method similar to printResult (sample.bal) to print this result as well.

**Review by anupama-pathirage — CHANGES_REQUESTED** [review]

File: component/samples/ethereum/sample.bal (hunk 5 above)
Comment (jaymine) [reply component/samples/ethereum/sample.bal]: No. It should be 2018. my mistake

File: component/src/org/wso2/ballerina/connectors/ethereum/ClientConnector.bal (hunk 2 above)
Comment (jaymine) [reply component/src/org/wso2/ballerina/connectors/ethereum/ClientConnector.bal]: Done!

File: component/samples/ethereum/sampleDashBoard.bal [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    int i = 0;
+    while (i < lengthof accounts) {
+        //get the account balance for each account
+        response, e = ethereumConnector.ethGetBalance(accounts[i].toString(), LATEST);
+        if (e != null) {
+            println(e);
+            return;
+        }
+        println(accounts[i].toString() + "\t" + response.result.toString());
```
Comment (anupama-pathirage) [inline component/samples/ethereum/sampleDashBoard.bal]: We can use printResult method here as well

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,330 @@
+<!--
+  ~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (anupama-pathirage) [inline pom.xml]: Incorrect year?

File: component/src/org/wso2/ballerina/connectors/ethereum/ClientConnector.bal [hunk 10]
```diff
@@ -0,0 +1,611 @@
...
+@Param {value:"jsonRPCVersion: JSON RPC version"}
+@Param {value:"id: Network id"}
+public connector ClientConnector (string URI, http:Options options, string jsonRPCVersion, int id) {
+
+    endpoint<http:HttpClient> ethereumEP {
+        create http:HttpClient(URI, options);
+    }
+
+    @Description {value:"Returns the current client version"}
+    @Return {value:"InResponse object"}
```
Comment (anupama-pathirage) [inline component/src/org/wso2/ballerina/connectors/ethereum/ClientConnector.bal]: Isn't it a Json? Check all the actions

File: component/findbugs-exclude.xml [hunk 11]
```diff
@@ -0,0 +1,19 @@
+<!--
+  ~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (anupama-pathirage) [inline component/findbugs-exclude.xml]: Incorrect year?

**Review by anupama-pathirage — CHANGES_REQUESTED** [review]

**Review by anupama-pathirage — APPROVED** [review]

**Pull request merged**